script/create-practice-repos

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env bash
+#
+# Create practice repos
+
+#################################################################
+# NOTE: You must have a personal access token (PAT)             #
+# saved to your environment variables to use this script.       #
+# We recommend a dedicated service account (e.g. githubteacher) #
+#################################################################
+
+# shellcheck source=script/shared_functions
+source ./shared_functions
+
+# shell variables
+collab_repo=$1
+template_name=$2
+practice_repo_name=$3
+org_url="https://$ROOT_URL/$CLASS_ORG"
+org_endpoint="https://$INSTANCE_URL/repos/$CLASS_ORG"
+template_url="https://github.com/githubtraining/$template_name"
+template_url_ghes="https://$ROOT_URL/$CLASS_ORG/$template_name"
+template_pages_url="https://$CLASS_ORG.github.io/$template_name"
+template_pages_url_ghes="https://$ROOT_URL/pages/$CLASS_ORG/$template_name"
+
+# https://superuser.com/a/1415376
+# Returns exit code 0 (success) if $1 is a reachable git remote url
+repo_is_reachable() {
+  local repo_url=$1
+  git ls-remote "$repo_url" CHECK_GIT_REMOTE_URL_REACHABILITY &>/dev/null
+}
+
+check_template_url() {
+  # if root url is not github.com
+  if [ "$ROOT_URL" != "github.com" ]; then
+    # if template can be found on GHES
+    if repo_is_reachable "$template_url_ghes"; then
+      # use template and GitHub Pages URL from GHES instead of the public template
+      template_url="$template_url_ghes"
+      template_pages_url="$template_pages_url_ghes"
+    # otherwise check if public template can be reached
+    elif repo_is_reachable "$template_url"; then
```

Comment:
Just noticed this part.  Is the separate URL scheme to allow for falling back to the public templates if users don't have local copies in GHES?

Reply:
Exactly! 😊 We’ll use the template on GHES if it exists, otherwise, we’ll try to use the public template.

Reply:
Do we want to work that into the initial class repo creation as well?  I know we treat the repo slightly differently than the personal practice repos, but it's still using the old logic from within the create-repo script via new-virtual.  Not sure if we include it in this PR or file an issue to address later.

Reply:
Hmm... Good point. Let’s address that in a different PR. I’ll create an issue.